File: Problem/Easy/118. Pascal's Triangle.py
# Optimal Solution --> O(numRows ^ 2)
class Solution:
    def generate(self, numRows: int) -> List[List[int]]:
        # First create and outer array
        outer = []
        
        # Then we create a for loop to loop with the inner array
        for i in range(numRows):
            
            # Inner array is created for each loop
            inner = []

            for j in range(i+1):

                if(j==0 or j==i):
                    inner.append(1)

                else:
                    x = outer[i-1][j] + outer[i-1][j-1]
                    inner.append(x)
            outer.append(inner)

        return outer

File: Problems/Easy/Java/118. Pascal's Triangle.java
// Brute Force

class Solution {
    public List<List<Integer>> generate(int numRows) {

        // First base case; if user requests zero rows, they get zero rows.
        if (numRows == 0) {
            return triangle;
        }

        List<List<Integer>> pascal = new ArrayList<>();

        for(int i=0; i<numRows; i++){
            List<Integer> col = new ArrayList<>();
            col.add(1);
            int count = i;
            for(int j=0; j<i; j++){
                if(count >= 2) {
                    int sum = pascal.get(i-1).get(j) + pascal.get(i-1).get(j+1); //NOTE: how to loop through arrays in arrays
                    col.add(sum);
                    count--;
                } else{
                    col.add(1);
                }
            }
            pascal.add(col);
        }
        
        return pascal;
    }
}